File: Real-Time-Search-Workshop/javascript/code/lab1.js
/**
 * @fileoverview Creates a node-redis client
 * @maker Joey Whelan
 */

import { createClient } from 'redis';
import * as dotenv from 'dotenv';
dotenv.config();

export class Lab1 {

    async run() {
        console.log('*** Lab 1 - Connect Client ***');
        const user = process.env.USER_NAME;
        const pwd = process.env.PASSWORD;
        const url = process.env.URL;
        const port = process.env.PORT;
    
        const client = createClient({url: `redis://${user}:${pwd}@${url}:${port}`});
        await client.connect();
        let result = await client.ping();
        console.log(result);
        return client;
    }
}

File: Real-Time-Search-Workshop/dotnet/code/lab1.cs
/**
 * @fileoverview Creates a StackExchange.Redis client
 * @maker Joey Whelan
 */
using System.Configuration;
using StackExchange.Redis;

namespace SearchWorkshop 
{
    public class Lab1 
    {
        public IDatabase Run() 
        {
            Console.WriteLine("*** Lab 1 - Connect Client ***");
            var redis = ConfigurationManager.AppSettings;
            ConfigurationOptions options = new ConfigurationOptions
            {
                EndPoints = {redis["URL"] + ":" + redis["Port"]},
                Password = redis["Password"]
            };
            ConnectionMultiplexer muxer = ConnectionMultiplexer.Connect(options);
            IDatabase client = muxer.GetDatabase();
            Console.WriteLine($"ping: {client.Ping().TotalMilliseconds} ms");
            return client;
        }
    }
}